### PR TITLE
Support for between operator 

### PR DESCRIPTION
**Pull request by twobytescy**
Utilizing GreaterThanOrEqualTo and LessThanOrEqualTo operators

### CODE REVIEW

File: DatabaseWrapper.Core/Expression.cs [hunk 1]
```diff
@@ -34,6 +35,24 @@ public Expression(object left, Operators oper, object right)
...
         }
+        /// <summary>
+        /// A structure in the form of term-operator-term that defines a boolean operation within a WHERE clause Supports only between operator.
+        /// </summary>
+        /// <param name="left">The left term of the expression; can either be a string term or a nested Expression.</param>
+        /// <param name="oper">The operator.</param>
+        /// <param name="right">The right term of the expression; Should be list of two values.</param>
+        public Expression(object left, Operators oper, List<object> right)
+        {
+            if (right.Count < 2) throw new ArgumentNullException(nameof(right));
```
Comment (jchristn) [inline DatabaseWrapper.Core/Expression.cs:46]: May I suggest:
```
if (right == null) throw new ArgumentNullException(nameof(right));
if (right.Count != 2) throw new ArgumentException("Right term must contain exactly two members.");
```

File: DatabaseWrapper.Core/Expression.cs [hunk 2]
```diff
@@ -34,6 +35,24 @@ public Expression(object left, Operators oper, object right)
...
+            if (oper != Operators.Between) throw new InvalidEnumArgumentException("Multiple values can only be used in Between operators");
```
Comment (jchristn) [inline DatabaseWrapper.Core/Expression.cs:47]: Please use: 
```
"Multiple values can only be used when the Operator is set to 'Between'."
```
(just for consistency!)

File: DatabaseWrapper.Core/Expression.cs [hunk 3]
```diff
@@ -34,6 +35,24 @@ public Expression(object left, Operators oper, object right)
...
+            Expression startOfBetween = new Expression(left, Operators.GreaterThanOrEqualTo, right.First());
+            Expression endOfBetween = new Expression(left, Operators.LessThanOrEqualTo, right.Last());
+            Expression e = PrependAndClause(startOfBetween, endOfBetween);
+            LeftTerm = e.LeftTerm;
+            Operator = e.Operator;
+            RightTerm = e.RightTerm;
+
```
Comment (jchristn) [inline DatabaseWrapper.Core/Expression.cs:54]: Please remove whitespace

**Comment by jchristn** [comment]
Thank you so much @twobytescy I'm pushing this to NuGet as we speak as version 3.0.1.  Cheers, Joel

**Comment by jchristn** [comment]
Hi @twobytescy in doing some testing I'm running into cases where the correct Expression constructor isn't being used.  I'm going to merge the explicit 'Between' constructor into the normal constructor and re-work this a bit.  Stay tuned.

**Comment by jchristn** [comment]
This seems to work:
```
            List<string> returnFields = new List<string> { "firstName", "lastName", "age" };
            List<object> vals = new List<object>();
            vals.Add(10);
            vals.Add(20);
            Expression e = new Expression("id", Operators.Between, vals);
            Console.WriteLine("Expression: " + e.ToString());
            _Database.Select("person", null, null, returnFields, e, "ORDER BY age DESC");
```
But this gets pulled into the regular ```public Expression(object left, Operators oper, object right)``` constructor instead of the desired ```public Expression(object left, Operators oper, List<object> right)``` constructor:
```
            List<string> returnFields = new List<string> { "firstName", "lastName", "age" }; 
            Expression e = new Expression("id", Operators.Between, new List<int> { 10, 20 });
            Console.WriteLine("Expression: " + e.ToString());
            _Database.Select("person", null, null, returnFields, e, "ORDER BY age DESC");
```

**Comment by jchristn** [comment]
So we may be stuck with the caller explicitly using ```List<object>``` even though they may want to use ```int```, ```decimal```, ```string```, etc...

https://stackoverflow.com/questions/62184219/multiple-constructors-using-generics-and-lists

Will watch this and see if there is a better solution, but for now, I think the one you've created is the most appropriate.  Caveat being the caller has to explicitly use ```List<object>```, but it works great when that's what they use.  

Cheers, Joel

**Comment by jchristn** [comment]
Hi @twobytescy given the confusion a consumer of the library may encounter when trying to do ```Between```, would you be opposed to me creating a static factory for this?  

i.e.
```
public static Expression Between(object left, List<object> right)
{
  ...
}
```

This would keep people from tripping over the issue of their Expressions not being built correctly because they use strongly-typed List<> instead of List<object> for the right term.

Let me know what you think.  Thanks!  ~Joel

**Comment by twobytescy** [comment]
Hello.

Well this was my initial implementation and I had it changed to be
consistent with your other constructors.

I totally agree with you

S

**Comment by jchristn** [comment]
My apologies @twobytescy i will get this changed and fixed on my end. Thanks!

**Pull request merged**